website/docs/language/style.mdx

```diff
@@ -0,0 +1,729 @@
+---
+page_title: Style Guide - Configuration Language
+description: >-
+  Learn recommended style conventions for the Terraform language and
+  workflows.
+---
+
+# Style Guide
+
+The flexibility of Terraform's configuration language gives you many options to choose from as you write your code, structure your directories, and test your configuration. We recommend that your organization adopt a style guide to apply to your Terraform development. While some decisions depend on your organization's needs or personal preferences, there are some common patterns that we suggest you adopt. These style suggestions will help keep your Terraform code consistent, legible, scalable, and maintainable. 
+
+This article discusses best practices, as well as some considerations to keep in mind as you develop your organization's style guide. The article is split into two sections. The first section covers code style recommendations, such as formatting and resource organization. The second section covers operations and workflow recommendations, such as lifecycle management through meta-arguments, versioning, and sensitive data management.
+
+## Code Style
+
+Writing Terraform code in a consistent style makes it easier to read and maintain. This section reviews code style patterns that we recommend that you adopt.
+
+- Run `terraform fmt` and `terraform validate` before committing your code to source control.
+- Use `#` for single and multi-line comments.
+- Use nouns for resource names and do not include the resource type in the name. 
+- Use underscores to separate multiple words in a resource name. Wrap resource type and name in double quotes in your resource definition.
+- Define dependent resources after those they depend on so that your code builds on itself.
+- Include a type and description for every variable.
+- Include a default value for optional variables.
+- Include a value and description for every output.
+- Use descriptive nouns to name your input variables, outputs, and local values and use underscores to separate multiple words. 
+- Avoid overuse of variables and local values.
+- Always include a default provider configuration.
+- Use `count` and `for_each` sparingly.
```

Comment:
I'm just a user, but this line is really ambiguous and vague. "Sparingly" implies rarely, if ever, use them, but I'm sure the majority of the community would agree on the exact opposite, and that the best practice is not to hard-code 10 of the exact same resources individually, which is what the style guide would imply if you say to use for_each "sparingly".

Does HCP really want to imply we should hard-code each individual resource instead of using for_each with a var?